doc/spec/namespace.md

```diff
@@ -0,0 +1,148 @@
+# Namespace
+The global namespace represents the full set of named repositories which can be
```

Comment:
The global namespace is the set of unique representations for any repository which may be referenced by a name. This would cover any possible image repository, not necessarily only public repositories which a given user has access to.

For example:
~~~
private.example.com/myrepo/myapp
docker.com/library/ubuntu
localhost:5000/some/image
~~~

All these may be considered as part of the global namespace.  The private repository at `private.example.com` may or may not be reachable by everyone, but it is always referencing the same repository. `private.example.com/` may be considered a scoped namespace for all images under that domain.  Within this scoped namespace all images are likely fetched from the same repository endpoints and use the same certificate authority to verify authenticity. `localhost` will be special cased scoped namespace as it is today to represent locally trusted images.

These scoped namespaces are likely an instance of distribution.  We plan to incorporate discovery endpoints as well to get the familiar behavior of spinning up a registry and referencing by its URL. It will be the responsibility of the distribution operator to setup the namespace accordingly.  We will also provide a tool such that the namespace may be configured locally without the need to do discovery. Authenticity restrictions may still apply if the namespace is locally misconfigured.

On top of describing all possible image repository names, this proposal is also attempting to describe a mechanism for separating the name from the registry transport. Current expansion rules would expand `ubuntu` to `docker.com/library/ubuntu` and trust will require that an image pulled as `docker.com/library/ubuntu` is actually that image.  However the transport may be configured to get this image from any mirror or cache, as long as the image is authentic.

This likely raised more questions but I hope this gave some clarity on what I intended here.  Let me know how I can make this document more clear.